watch.go

```diff
@@ -61,16 +61,15 @@ func (w WatchSet) Watch(timeoutCh <-chan time.Time) bool {
 		}
 	}()
 
-	return w.WatchCtx(ctx)
+	return w.WatchCtx(ctx) == context.Canceled
```

Comment:
This should probably be `!= nil` to cover deadline exceeded errors, too.

Reply:
In this case we are actually cancelling the context so this is the only expected error.